specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1182 @@
+
+notification.Progress(data); // Updates the AppNotification values
+winrt::AppNotificationManager::Default().Show(notification);
+```
+
+Example result: ![Progress Bar Example 1](ProgressBarExample1.png)
+
+Then update the ProgressBar using AppNotificationProgressData:
+
+```c#
+data.SequenceNumber(2);
+data.Title(L"Katy Perry"); // Binds to {progressTitle} in xml payload
+data.Value(1.0); // Binds to {progressValue} in xml payload
+data.ValueStringOverride(L"2/2 songs"); // Binds to {progressValueString} in xml payload
+data.Status(L"Downloaded!"); // Binds to {progressStatus} in xml payload
+
+auto result = co_await winrt::Microsoft::Windows::AppNotifications::AppNotificationManager::Default().UpdateAsync(data, L"Tag", L"Group");
+```
+
+Example result: ![Progress Bar Example 2](ProgressBarExample2.png)
+
+# Retrieving Arguments
+AppNotificationContent and Button return arguments to the activated application when the user clicks on the component. Developers can
+add multiple key/value pairs and Windows App SDK will serialize them into a string.
+
+Below is an example with AppNotificationContent and Button:
+
+```c#
+AppNotificationContent(
+    .AddArgument(L"action", L"appNotificationBodyTouch")
+    .AddArgument(L"user", 938163)
+    .AddButton(Button(L"Send")
+        .AddArgument(L"action", L"buttonTouch")
+        .AddArgument(L"user", 938163)));
+```
+
+When the user clicks on the AppNotification body, the resulting argument string is:
+- "action=appNotificationBodyTouch;user=938163"
+
+When the user clicks on the Button, the result argument string is:
+- "action=buttonTouch;callId=938163"
+
+Notice that ';' is used as the delimiter between each argument.
+
+To retrieve the arguments from the serialized string, developers will use
+AppNotificationReceivedEventArgs::Arguments to get the values from the argument string as a map.
+
+Below is an example usage:
+```c#
+using namespace winrt::Microsoft::Windows::AppLifecycle;
+using namespace winrt::Microsoft::Windows::AppNotifications;
+
+// Retrieve the activation args using Windows App SDK AppLifecycle functions
+AppActivationArguments args { AppLifecycle::AppInstance::GetCurrent().GetActivatedEventArgs() };
+ExtendedActivationKind kind { args.Kind() };
+
+// Check the activation kind
+if (kind == ExtendedActivationKind::AppNotification)
+{
+    AppNotificationActivatedEventArgs appActivatedArgs { args.Data().as<AppNotificationActivatedEventArgs>() };
+
+    // Retrieve the serialized arguments from AppNotificationActivatedEventArgs
+    winrt::Windows::Foundation::Collections::IMap<winrt::hstring, winrt::hstring> argumentMap { appActivatedArgs.Argument() };
+}
+```
+# Full API Details
+
+```c#
+namespace Microsoft.Windows.AppNotifications.Builder
+{
+    runtimeclass TextProperties
+    {
+        // Contains the set of <text> attributes
+        TextProperties();
+
+        TextProperties SetLanguage(String language);
+        TextProperties UsesCallScenarioAlign();
```

Comment:
Shoud this be UsesCallScenarioAlign(); or UseCallScenarioAlign(); (without the s in Use)?

Reply:
I decided on Use in my impl, I'll update it